### PR TITLE
feat(consul): new package

### PR DESCRIPTION
**Pull request by oljekechoro**

### CODE REVIEW

File: packages/consul/package.json [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    "lint": "eslint src/**/*.ts",
+    "lint:fix": "yarn lint --fix",
+    "clean": "rimraf target typings buildcache docs flow-typed",
+    "build": "yarn build:es5 && yarn build:es6 && yarn build:ts && yarn build:libdef && yarn docs && yarn uglify",
+    "build:es5": "mkdir -p target/es5 && tsc -p tsconfig.es5.json",
+    "build:es6": "mkdir -p target/es6 && tsc -p tsconfig.es6.json",
+    "build:ts": "cp -r src/main/ts/ target/ts/",
+    "build:bundle": "microbundle build src/main/ts/index.ts -o target/bundle",
+    "build:libdef": "libdefkit --tsconfig=tsconfig.es5.json --tsconfig=tsconfig.es6.json",
+    "docs": "typedoc --readme README.md --tsconfig tsconfig.json src/main --ignoreCompilerErrors || exit 0",
```
Comment (antongolub) [inline packages/consul/package.json:37]: * Use typedoc 0.22 beta
* Update `typedoc.json`
* rm typedoc-plugin-external-module-name plugin

**Review by antongolub — CHANGES_REQUESTED** [review]

File: packages/consul/package.json (hunk 1 above)
Comment (oljekechoro) [reply packages/consul/package.json:37]: maybe ^0.20.0-beta.27?

Comment (antongolub) [reply packages/consul/package.json:37]: Yep.

File: README.md [hunk 2]
```diff
@@ -1,13 +1,18 @@
 # QIWI Nestjs enterprise
 Components to build Nestjs-based applications for enterprise
 
-* [@qiwi/nest-enterprise-common](./packages/common/README.md)
+* [@qiwi/nestjs-enterprise-common](./packages/common/README.md)
```
Comment (antongolub) [inline README.md:4]: I suggest using absolute links, so they can be usable on the npm site too.

**Review by antongolub — CHANGES_REQUESTED** [review]

File: packages/consul/src/main/ts/consul.service.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
+import { Inject, Injectable } from '@nestjs/common'
+import {
+  ConsulDiscoveryService,
+  IConsulKvSetOptions,
+  INormalizedConsulKvValue,
+} from '@qiwi/consul-service-discovery'
+import { IConfig, ILogger, IPromise } from '@qiwi/substrate-types'
```
Comment (antongolub) [inline packages/consul/src/main/ts/consul.service.ts:7]: @qiwi/substrate

File: packages/consul/src/main/ts/consul.service.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+  async register() {
+    const serviceName: string = this.config.get('name')
+    const consulHost: string = this.config.get('consul.host')
+    const consulPort: string = this.config.get('consul.port')
+    const tags: string[] = this.config.get('consul.tags')
+    const consulToken: string = this.config.get('consul.token')
+    const consulUrl = `http://${consulHost}:${consulPort}/v1/agent/service/register`
+    const port: number = this.config.get('server.port')
+    const ip: string = this.config.get('server.host')
```
Comment (antongolub) [inline packages/consul/src/main/ts/consul.service.ts:39]: const consulCheckRegInterval = this.config.get('consul.checkRegInterval') || CONSUL_CHECK_REG_INTERVAL

**Review by antongolub — CHANGES_REQUESTED** [review]

**Review by antongolub — APPROVED** [review]
LGTM

**Pull request merged**